### PR TITLE
Alignment of activity metrics into Growth Maturity and Decline

### PR DESCRIPTION
**Pull request by sgoggins**
A number of activity metrics were abandon in our reorganization into categorical working groups. A number of the GMD related ones have been added to that page in this pull request. 

Worth noting is that some of the metrics are GitHub specific. Forks, for example. I suggest we address distinguishing these kinds of repository specific metrics at a later time.

### CODE REVIEW

**Comment by GeorgLink** [comment]
​Will there be ever a 'stable' version? I'm okay with this document in the
metrics repo to represent a work-in-progress version. Real work happens in
pull-requests and semi-stable versions are merged. Capturing all work in
one place also reduces uncertainty over which version is authoritative.

**Comment by jgbarah** [comment]
> Will there be ever a 'stable' version?

Good point! I think this will be really an "evolving" stable version. We could mimic what we do in many FOSS projects: continuous release in the wg-gmd repository (prs, issues, etc. happen there), with copy to this repo when we decide we have a version which is "stable". For example, now we started with wg-gmd, but say that in two months, we have reviewed all metrics, and we release what we could call "GMD 1.0". That could come to this repo, so that this is is the "authoritative" CHAOSS GMD 1.0 release (until we produce 1.1 or 2.0...).

**Comment by germonprez** [comment]
I can see advantages to both. 

1) Work in the main metrics repo certainly does reduce work occurring in two places. Clarity is key. 

2) Work in the GMD repo as released to the metrics repo allows the work to happen in a messy way in the workgroup without any real concerns for having to backtrack on anything in the metrics repo. 

The question, I think, becomes where in the workflow do we connect the GMD workgroup to the Metrics committee? Point 1 is earlier. Point 2 is later.

**Comment by sgoggins** [comment]
I did this in metrics because I don’t think it’s the work of the gmd committee so much as cleaning up metrics that were abandon when we switched to the workgroup model. 

Sean P. Goggins
(816) 368-1364
http://www.seangoggins.net
visit: http://www.sociotech.net
visit: http://www.datafactories.org 

"I force people to have coffee with me, just because I don't trust that a friendship can be maintained without any other senses besides a computer or cellphone screen.”
— John Cusack


On May 18, 2018, at 06:07, Matt Germonprez <notifications@github.com> wrote:

I can see advantages to both.

Work in the main metrics repo certainly does reduce work occurring in two places. Clarity is key.

Work in the GMD repo as released to the metrics repo allows the work to happen in a messy way in the workgroup without any real concerns for having to backtrack on anything in the metrics repo.

The question, I think, becomes where in the workflow do we connect the GMD workgroup to the Metrics committee? Point 1 is earlier. Point 2 is later.

**Comment by sgoggins** [comment]
@germonprez : I think that once the GMD committee is into its own workflow, we maybe focus on the "GMD" branch/repo.  Right now I think we are completing an overall realignment of the metrics and assignment to working groups at the 'overall' CHAOSS project level.

**Comment by sgoggins** [comment]
@jgbarah : I'm thinking it will be helpful to "unorphan" the orphaned activity metrics and then dive into the repo for GMD. That way we're all talking about the same base.  In my mind, which could be clouded by expediency, the pull request is aimed at updating the "whole of CHAOSS" and "cleaning up the metrics repo" more than the work of the GMD group.  

I am comfortable with whatever choice folks in the group want to make.  If you prefer taking this to GMD, then I will do that. Let me know what you think of my rationale and if you agree.

File: 2_Growth-Maturity-Decline.md [hunk 1]
```diff
@@ -4,30 +4,41 @@ Goal: Identify if the level of maturity the project has reached and whether it i
...
+[Code Review Iteration](activity-metrics/code-review-iteration.md) | What is the number of iterations that occur before a merge request is accepted or declined?
+[Forks](activity-metrics/forks.md) | Forks are a concept in distributed version control systems like GitHub. It is a proxy for the approximate number of developers who have taken a shot at building and deploying the codebase *for development*.
+[Pull Requests Open](activity-metrics/pull-requests-open.md) | Number of open pull requests.
+[Pull Requests Closed](activity-metrics/pull-requests-made-closed.md) | Number of closed pull requests.
+[Pull Request Comment Duration](activity-metrics/pull-requests-comment-duration.md) | The difference between the timestamp of the pull request creation date and the most recent comment on the pull request.
+[Pull Request Comment Diversity](activity-metrics/pull-requests-comment-diversity.md) | Number of each people discussing each pull request.
+[Pull Request Comments](activity-metrics/pull-request-comments.md) | Number of comments on each pull request. 
+
+
+[](activity-metrics/)
```
Comment (GeorgLink) [inline 2_Growth-Maturity-Decline.md]: Line 41 appears to be degenerated.
> [](activity-metrics/)

File: 2_Growth-Maturity-Decline.md [hunk 2]
```diff
@@ -4,30 +4,41 @@ Goal: Identify if the level of maturity the project has reached and whether it i
...
+[Code Review Iteration](activity-metrics/code-review-iteration.md) | What is the number of iterations that occur before a merge request is accepted or declined?
+[Forks](activity-metrics/forks.md) | Forks are a concept in distributed version control systems like GitHub. It is a proxy for the approximate number of developers who have taken a shot at building and deploying the codebase *for development*.
+[Pull Requests Open](activity-metrics/pull-requests-open.md) | Number of open pull requests.
+[Pull Requests Closed](activity-metrics/pull-requests-made-closed.md) | Number of closed pull requests.
+[Pull Request Comment Duration](activity-metrics/pull-requests-comment-duration.md) | The difference between the timestamp of the pull request creation date and the most recent comment on the pull request.
+[Pull Request Comment Diversity](activity-metrics/pull-requests-comment-diversity.md) | Number of each people discussing each pull request.
+[Pull Request Comments](activity-metrics/pull-request-comments.md) | Number of comments on each pull request.
```
Comment (GeorgLink) [inline 2_Growth-Maturity-Decline.md]: Added metrics: 

* Forks
* Pull Request Open
* Pull Request Closed
* Pull Request Comment Duration
* Pull Request Comment Diversity
* Pull Request Comments

Currently do not follow the GQM approach. I suggest rewriting the description as a question.

**Comment by sgoggins** [comment]
THis seems superceded by work since originally made.

**Pull request closed without merging**